hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/format/RecordIterators.java

```diff
@@ -50,8 +50,7 @@ public static ClosableIterator<RowData> getParquetRecordIterator(
       Path path,
       long splitStart,
       long splitLength) throws IOException {
-    InternalSchema fileSchema = internalSchemaManager.getFileSchema(path.getName());
```

Comment:
Why did you move this code into else branch?
`getFileSchema()` may return empty schema which allow us avoiding  unnecessary creation of `ParquetSplitRecordIterator`

```java
  if (querySchema.equals(fileSchema)) {
    return InternalSchema.getEmptyInternalSchema();
  }
```

Reply:
My bad, failed to account for this case. Let me revert this + add a few more comments.

Reply:
@trushev Thank you for helping to review this. I've reverted the changes and added more javadocs. Please help to take a look + review it again.

Reply:
LGTM, thank you for the improvement